examples/distributed_tree/distributed_knn.cpp

```diff
@@ -0,0 +1,93 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <cstdarg>
+#include <cstdio>
+#include <iostream>
+#include <random>
+#include <vector>
+
+#include <mpi.h>
+
+using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+using MemorySpace = ExecutionSpace::memory_space;
+
+namespace Example
+{
+template <class Points>
+struct Nearest
+{
+  Points points;
+  int k;
+};
+template <class Points>
+Nearest(Points const &, int) -> Nearest<Points>;
+
+struct IndexAndRank
+{
+  int index;
+  int rank;
+};
```

Comment:
We never documented the distributed, huh. It seems weird that we need to define the return structure in the user code, we should provide it in ArborX.

Reply:
I agree